dbt_platform_helper/utils/aws.py

```diff
@@ -349,6 +350,46 @@ def get_postgres_connection_data_updated_with_master_secret(session, parameter_n
     return parameter_data
 
 
+def get_redis_supported_versions(elasticache_client=None):
+
+    supported_versions = []
+
+    if not elasticache_client:
+        session = get_aws_session_or_abort()
+        elasticache_client = session.client("elasticache")
```

Comment:
I don't think we have any tests for this. Instead of adding tests, can we just delete these three lines?

Reply:
They exist for the tests solely - else it is not easy to inject a stubbed boto3 client.

Reply:
I suspect this may be solved by extracting to an AWS provider class. Worth speaking to the others about what they are doing.

Reply:
Move to https://github.com/uktrade/platform-tools/pull/624/files#r1841894953 because this comment is outdated for some reason.